### PR TITLE
Add Cambridge latex style resources

### PR DESCRIPTION
**Pull request by cheungdaven**
*Description of changes:*

(1) Rewrite some sphinx default latex functions and styles.
(2) The cambridge latex style is downloaded automatically. Here, I photoshopped two images (included in latex_style directory) based on alex's suggestions. 
(3) The https://github.com/d2l-ai/d2l-book/pull/56 of d2l-book is dependent on this pr.

To ensure the smoothness of the qr code generation process, please do not use any special characters in the URL, otherwise, it won't go through. You can easily replace the special characters with URL encoding. please refer to: https://www.urlencoder.io/learn/ 

To use this style, please specify the style in the config.ini of d2l-en.
style = cambridge

By submitting this pull request, I confirm that you can use, modify,
copy, and redistribute this contribution, under the terms of your
choice.

### CODE REVIEW

**Comment by astonzhang** [comment]
Since the current d2lbook does not support `d2lbook build pdf --style`, could you specify the dependency PR in d2lbook, if any?

**Comment by cheungdaven** [comment]
> Since the current d2lbook does not support `d2lbook build pdf --style`, could you specify the dependency PR in d2lbook, if any?

This pr depends on [PR56](https://github.com/d2l-ai/d2l-book/pull/56) of the d2lbook repository.

**Comment by cheungdaven** [comment]
@astonzhang Now all cambridge latex resources are downloaded and extracted through the S3 link.

**Comment by astonzhang** [comment]
Thanks. For better readability, could you add more comments on those `*.sty` files to describe all the customized styles that are not generated by Sphinx? It may also be helpful to specify where the customized styles come from, such as from PT1

File: config.ini [hunk 1]
```diff
@@ -57,13 +57,15 @@ include_css = static/d2l.css
...
 post_latex = ./static/post_latex/main.py
+latex_url = https://d2l-webdata.s3.us-west-2.amazonaws.com/latex-styles/PT1.zip
 
 latex_logo = static/logo.png
 main_font = Source Serif Pro
 sans_font = Source Sans Pro
 mono_font = Inconsolata
 
 bibfile = d2l.bib
+style = default
```
Comment (astonzhang) [inline config.ini:68]: Can you add new entries `latex_url` and `style` to https://github.com/d2l-ai/d2l-book/blob/master/d2lbook/config_default.ini? Please set `style = default` in config_default.ini so we can remove this line.

**Comment by cheungdaven** [comment]
@d2l-bot  rebuild

**Comment by astonzhang** [comment]
Great job, thanks!
This PR has been tested with and without `style = cambridge`.

**Pull request merged**